MetaMask/metamask-mobile
# fix: fix walletconnect source validation

## **Description**
This PR fixes a request source validation check for walletconnect. We previously checked if the metadata url matches the origin. However the origin could be `wc::https://appkit-lab.reown.com/` or `wc::https://opensea.io` for example.

## **Related issues**

Fixes: [SDK-56](https://consensyssoftware.atlassian.net/browse/SDK-56)

## **Manual testing steps**

1. Go to [WallectConnect Dapp](https://appkit-lab.reown.com/)
2. Connect using MetaMask
3. Perform a transaction
4. Check the MetaMask mobile analytics on Segment
5. The `request_source` should be `WalletConnect`

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[SDK-56]: https://consensyssoftware.atlassian.net/browse/SDK-56?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
